### PR TITLE
:sparkles: Source Bing Ads: Accounts: update docs, unit tests for empty streams

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
Bing Ads Certification: 
- unittests for empty streams
- pk info in docs for streams with user-defined pk 

## How
- mocked response for empty stream to test stream behaviour
- updated docs with info about how to define pk for streams with user-defined pk

### CODE REVIEW

**Review by maxi297 — COMMENTED** [review]
Thanks for adding the tests! I have a couple question regarding the changes to `Accounts`

File: airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py [hunk 1]
```diff
@@ -193,6 +195,50 @@ def request_params(
...
         }
 
+    @property
+    def state(self) -> Mapping[str, Any]:
+        return self._state
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        current_state_value = self._state.get(self.cursor_field, "")
+        if record_state_value := value[self.cursor_field]:
```
[the remark this answers is not in the document]
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py:205]: for first record cursor value is not set

File: airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py [hunk 2]
```diff
@@ -193,6 +195,50 @@ def request_params(
...
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+        **kwargs: Mapping[str, Any],
+    ) -> Iterable[Mapping[str, Any]]:
+        stream_state = stream_state or {}
+        next_page_token = None
+        account_id = str(stream_slice.get("account_id")) if stream_slice else None
```
[the remark this answers is not in the document]
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py]: yes, you are right, account_id and customer_id are redundant here. I removed it

File: airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py [hunk 3]
```diff
@@ -193,6 +195,50 @@ def request_params(
...
+        customer_id = str(stream_slice.get("customer_id")) if stream_slice else None
+
+        while True:
+            params = self.request_params(
+                stream_state=stream_state,
+                stream_slice=stream_slice,
+                next_page_token=next_page_token,
+                account_id=account_id,
```
[the remark this answers is not in the document]
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py]: it must be none for accounts stream, i just have missed it

File: airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py [hunk 4]
```diff
@@ -193,6 +195,43 @@ def request_params(
...
+        stream_state = stream_state or {}
+        next_page_token = None
+
+        while True:
+            params = self.request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+            response = self.send_request(params, customer_id=None)
+            for record in self.parse_response(response):
+                record = self.transform(record, stream_slice)
+                if sync_mode == SyncMode.incremental:
+                    if self.state.get(self.cursor_field, "") <= record.get(self.cursor_field):
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py:225]: I think this logic does not work. Records are not sorted so we can assume the sequence of records:
* "LastModifiedTime": "2023-01-01T00:00:00.000000"
* "LastModifiedTime": "2020-01-01T00:00:00.000000"

Given an initial sync and this sequence of accounts, the second record would not be synced. Hence, I think we might have underestimated the complexity of implementing semi-incremental. I think we can exclude this for the certification and see if we implement it if a user request it.

**Review by maxi297 — COMMENTED** [review]
Let's keep the tests which add a very nice safety net but remove the change to make Accounts incremental. Once the incremental part is reverted, I'll approve

File: airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py (hunk 4 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py:225]: I agree, I will remove incremental from this pr

**Review by lazebnyi — APPROVED** [review]
lgtm

**Review by maxi297 — APPROVED** [review]
LGTM! Thanks for adding this safety net

**Review by katmarkham — APPROVED** [review]

**Pull request merged**